src/java.desktop/share/classes/sun/font/StrikeCache.java

```diff
@@ -98,40 +105,161 @@ public final class StrikeCache {
+        return (char)heightHandle.get(seg);
+    }
+
+    @SuppressWarnings("restricted")
+    static final char getGlyphRowBytes(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return (char)rowBytesHandle.get(seg);
+    }
+
+    @SuppressWarnings("restricted")
+    static final byte getGlyphManaged(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return (byte)managedHandle.get(seg);
+    }
+
+    @SuppressWarnings("restricted")
+    static final float getGlyphTopLeftX(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return (float)topLeftXHandle.get(seg);
+    }
+
+    @SuppressWarnings("restricted")
+    static final float getGlyphTopLeftY(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return (float)topLeftYHandle.get(seg);
+    }
+
+    @SuppressWarnings("restricted")
+    static final long getGlyphCellInfo(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return ((MemorySegment)cellInfoHandle.get(seg)).address();
+    }
+
+    @SuppressWarnings("restricted")
+    static final void setGlyphCellInfo(long ptr, long val) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        MemorySegment segval = MemorySegment.ofAddress(val);
+        cellInfoHandle.set(seg, segval);
+    }
+
+    @SuppressWarnings("restricted")
+    static final long getGlyphImagePtr(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        return ((MemorySegment)imageHandle.get(seg)).address();
+    }
+
+    @SuppressWarnings("restricted")
+    static final MemorySegment getGlyphPixelData(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        char hgt = (char)heightHandle.get(seg);
+        char rb = (char)rowBytesHandle.get(seg);
+        MemorySegment pixelData = (MemorySegment)imageHandle.get(seg);
+        pixelData = pixelData.reinterpret(rb * hgt);
+        return pixelData;
+    }
+
+    @SuppressWarnings("restricted")
+    static final byte[] getGlyphPixelBytes(long ptr) {
+        MemorySegment seg = MemorySegment.ofAddress(ptr);
+        seg = seg.reinterpret(GLYPHIMAGESIZE);
+        char hgt = (char)heightHandle.get(seg);
+        char rb = (char)rowBytesHandle.get(seg);
+        MemorySegment pixelData = (MemorySegment)imageHandle.get(seg);
+        int sz = rb * hgt;
+        pixelData = pixelData.reinterpret(sz);
+        byte[] bytes = new byte[sz];
+        MemorySegment.copy(pixelData, ValueLayout.JAVA_BYTE, 0, bytes, 0, sz);
+        return bytes;
```

Comment:
You could use `toArray` here:

```suggestion
        return pixelData.toArray(ValueLayout.JAVA_BYTE);
```